src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -675,16 +677,29 @@ internal static string AppendChangesToDescription(string prDescription, Build ol
                     prLink = $@"- [{comment}]({repoURL}/commit/{fullSHA})";
                 }
 
-                description.AppendLine(prLink);
+                const string limitMessage = "Changelog truncated due to description length limit.";
 
-                if (description.Length > 3500)
+                // we want to be able to fit this PR link, as well as the limit message (plus line breaks) in case the next PR link doesn't fit
+                int limit = hardLimit - (prLink.Length + 1) - (limitMessage.Length + 1);
+                if (description.Length > limit)
                 {
-                    description.AppendLine("Changelog truncated due to description length limit.");
+                    description.AppendLine(limitMessage);
                     break;
                 }
+                else
+                {
+                    description.AppendLine(prLink);
+                }
             }
 
-            return description.ToString();
+            var result = description.ToString();
+            if (result.Length > hardLimit)
```

Comment:
Will this ever be true if we are breaking before we hit the hard limit?

Reply:
I don't expect to really hit this. I added it as more of a guardrail, so if the insertion pipeline ever fails, we might be able to diagnose it more easily